x-pack/platform/plugins/shared/streams/common/query.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const getKqlAsCommandArg = (str: string): string => {
```

Comment:
We shouldn't start this kind of thing - the esql ast package should have all the helpers to do this in a clean way that is not maintained by us: https://github.com/elastic/kibana/blob/313f973aecac80f7eb6340ba517870999e0da1a3/src/platform/packages/shared/kbn-esql-ast/src/builder/README.md#L18-L19

With this tool you can also replace the string-based construction of the ESQL query in `discover_helpers`. I know we don't use it across the streams code base, but we should (precisely for things like escaping stuff correctly)

Reply:
Fixed in https://github.com/elastic/kibana/pull/228803/commits/aaa13560f23172ad48cce8fa913430ecb089b49c